### ISSUE TITLE
AutoEncoder shows poor performance for iris data set

### ISSUE THREAD

**Issue by lucidfrontier45** [aigamedev/scikit-neuralnetwork#103]
Hi. I followed this document
http://scikit-neuralnetwork.readthedocs.org/en/latest/guide_advanced.html#unsupervised-pre-training
ant tried to apply 3-layer MLP with pre-trained autoencoder  to the iris data set.

The accuracy was too low, only 0.333 even for training data set.
Below is my code. Am I missing something?

Thank you.

```python
from sklearn.datasets import load_iris
from sknn.mlp import Classifier, Layer as NNLayer
from sknn.ae import AutoEncoder, Layer as AELayer

iris = load_iris()
X = iris.data
y = iris.target

ae = AutoEncoder(layers=[
    AELayer("Tanh", units=10),
    AELayer("Sigmoid", units=2)
], learning_rate=0.002, n_iter=10).fit(X)

model = Classifier(layers=[
    NNLayer("Tanh", units=10),
    NNLayer("Sigmoid", units=2),
    NNLayer("Softmax")
], learning_rate=0.001, n_iter=50)

ae.transfer(model)

model.fit(X, y)

print(model.score(X, y))
```

**Comment by alexjc** [comment]
Try the usual. More iterations? Higher learning rate? Better learning rule?

The code for auto-encoder I'm sure is much less mature in PyLearn2 than the regular NN.  For example you can use modern activations like ReLU, not possible with AE.  Try doing an auto-encoder yourself with a regression NN that uses input as output.

**Comment by lucidfrontier45** [comment]
@alexjc 

Thank you for reply. I first intended to write AE by Regresser but I couldn't found how to access the learned coefficients (`W` and `b` in most literature) of hidden layers.  Is there any API to obtain hidden layer parameters?

**Comment by alexjc** [comment]
All neural networks have a way to get weights, but you're right it's not well documented and the API hasn't been designed.  Do a search for `_mlp_get_weights` or something like that. It's used it `__getstate__` for serialization.

It's quite busy this week but I can look into it in about 10 days...

**Comment by lucidfrontier45** [comment]
OK, I'll take a look at `_mlp_get_weights`.
